databricks/koalas/indexes.py

```diff
@@ -1312,6 +1383,92 @@ def levshape(self):
         return tuple(result)
 
+    def to_frame(self, index=True, name=None) -> DataFrame:
+        """
+        Create a DataFrame with the levels of the MultiIndex as columns.
+        Column ordering is determined by the DataFrame constructor with data as
+        a dict.
+
+        Parameters
+        ----------
+        index : boolean, default True
+            Set the index of the returned DataFrame as the original MultiIndex.
+        name : list / sequence of strings, optional
+            The passed names should substitute index level names.
+
+        Returns
+        -------
+        DataFrame : a DataFrame containing the original MultiIndex data.
+
+        See Also
+        --------
+        DataFrame
+
+        Examples
+        --------
+        >>> tuples = [(1, 'red'), (1, 'blue'),
+        ...           (2, 'red'), (2, 'blue')]
+        >>> idx = ks.MultiIndex.from_tuples(tuples, names=('number', 'color'))
+        >>> idx  # doctest: +SKIP
+        MultiIndex([(1,  'red'),
+                    (1, 'blue'),
+                    (2,  'red'),
+                    (2, 'blue')],
+                   names=['number', 'color'])
+        >>> idx.to_frame()  # doctest: +NORMALIZE_WHITESPACE
+                      number color
+        number color
+        1      red         1   red
+               blue        1  blue
+        2      red         2   red
+               blue        2  blue
+
+        By default, the original Index is reused. To enforce a new Index:
+
+        >>> idx.to_frame(index=False)
+           number color
+        0       1   red
+        1       1  blue
+        2       2   red
+        3       2  blue
+
+        To override the name of the resulting column, specify `name`:
+
+        >>> idx.to_frame(name=['n', 'c'])  # doctest: +NORMALIZE_WHITESPACE
+                      n     c
+        number color
+        1      red    1   red
+               blue   1  blue
+        2      red    2   red
+               blue   2  blue
+        """
+        if name is None:
+            name = [name if name is not None else (str(i),)
+                    for i, name in enumerate(self._internal.index_names)]
+        elif is_list_like(name):
+            if len(name) != len(self._internal.index_map):
+                raise ValueError("'name' should have same length as number of levels on index.")
+            name = [n if isinstance(n, tuple) else (n,) for n in name]
+        else:
+            raise TypeError("'name' must be a list / sequence of column names.")
+
+        sdf = self._internal.sdf.select([scol.alias(name_like_string(idx))
+                                         for scol, idx in zip(self._internal.index_scols, name)]
+                                        + [NATURAL_ORDER_COLUMN_NAME])
+
+        if index:
+            index_map = [(name_like_string(idx), n)
+                         for idx, n in zip(name, self._internal.index_names)]
+        else:
+            index_map = None  # type: ignore
```

Comment:
I'm not pretty sure,

but how about `index_map = [(SPARK_INDEX_NAME_FORMAT(0), None)]`??

then maybe can we avoid `attach_default_index` ?

Reply:
What's the `SPARK_INDEX_NAME_FORMAT(0)` here then?

Reply:
ah, i was mistaken. sorry for the confusion. 😞